webapp/app/forms/validators.py

```diff
@@ -88,6 +88,17 @@ def __call__(self, form, field):
             raise ValidationError(_('validate.invalid-idnr'))
 
 
+class ValidUnlockCode:
+    def __call__(self, form, field):
+        input_str = str(field.data)
+        # must contain 14 digits
+        if len(input_str) != 14:
+            raise ValidationError(_('validate.unlock-code-length'))
+        # three block separated by '-'
+        if input_str[4] != '-' or input_str[9] != '-':
```

Comment:
couldn't we use `_value` for this and then just check the length?

Reply:
Yes. But as _value is a protected attribute only used for displaying (in my opinion it should theoretically not even be used in the widget but WTForms does it themselves), I think data is better suited here. Plus we can make sure that we're actually sending the correct data, not only show it correctly.